BUILDING.md

```diff
@@ -1,40 +1,59 @@
 ### Building
 
-To build coremltools from source, you require you need
-[CMake](https://cmake.org) and
-[Miniconda](https://docs.conda.io/en/latest/miniconda.html) to configure the
-project.
+This page describes building Core ML Tools (coremltools) from the source repository.
 
-Our makefile & scripts require the **zsh** shell (default shell for macOS
-10.16+) installed in `/usr/bin`.
+## Requirements
 
-The following targets will handle the development environment for you. If you
-need to add packages, edit the reqs/pip files and the auto-environment will
-install them automatically.
+To build coremltools from source, you need the following:
 
+* [CMake](https://cmake.org/)
+* [Miniconda](https://docs.conda.io/en/latest/miniconda.html) (or [Anaconda](https://www.anaconda.com/))
+* [Zsh shell](http://zsh.sourceforge.net/) (the default shell for macOS 10.16+) installed in /usr/bin
 
-* `build` | Build coremltools in *debug* mode (include symbols).
+## Build script
+
+Follow these steps:
+
+1. Fork and clone the GitHub [coremltools repository](https://github.com/apple/coremltools).
+2. Run the [build.sh](scripts/build.sh) script to activate the appropriate release environment and build `coremltools`. The build.sh script uses Python 3.7, but you can include `--python=3.5` (or `3.6`, `3.8`, and so on) as a argument to change the Python version. The script creates a new `build` folder with the coremltools distribution, and a `dist` folder with Python wheel files.
+3. Run the [test.sh](scripts/test.sh) script to test the build.
+
+Both scripts automatically run the `env_create` and `env_activate` scripts. To examine them, see the following:
+
+* [`env_create.sh`](scripts/env_create.sh): Reads the required packages in `coremltools/reqs` and installs them into a new environment. This new environment is located at `<repo root>/coremltoos/envs`. The environment is named after the `py` parameter. For example, a development environment with py 3.7 is named `coremltools-dev-py37`.
+* [`env_activate.sh`](scripts/env_activate.sh): Activates the new environment with `conda activate`.
```

Comment:
I activate the environment by `source  scripts/env_activate.sh`. That also what the help for that script say too.

I think `conda activate` just activates your default Conda environment.

Reply:
OK, changed to the following:

1. Fork and clone the GitHub [coremltools repository](https://github.com/apple/coremltools).

2. Run the [build.sh](scripts/build.sh) script to build `coremltools`. 
	* By default this script uses Python 3.7, but you can include `--python=3.5` (or `3.6`, `3.8`, and so on) as a argument to change the Python version. 
	* The script creates a new `build` folder with the coremltools distribution, and a `dist` folder with Python wheel files.
	
3. Run the [test.sh](scripts/test.sh) script to test the build.

**Under the hood**: If an Anaconda or Miniconda environment doesn't already exists or is not up-to-date, the `build.sh` script automatically runs the [`env_create.sh`](scripts/env_create.sh) script to create the environment. It then uses [`env_activate.sh`](scripts/env_activate.sh) to activate the environment and set up the appropriate version of Python. The new environment is located at `<repo root>/coremltoos/envs` and is named after the `py` parameter. For example, a development environment with py 3.7 is named `coremltools-dev-py37`.